src/main/java/org/matsim/contrib/gtfs/GtfsConverter.java

```diff
@@ -1,162 +1,226 @@
-			tl.getAttributes().putAttribute("gtfs_route_type", String.valueOf(route.route_type)); // route type is a required field according to GTFS specification
-			
-			String routeShortName = null;
-			if (route.route_short_name != null) {
-				routeShortName = route.route_short_name;
-			} else {
-				// use id in case there is no route short name
-				routeShortName = String.valueOf(route.route_id);
-			}
-			tl.getAttributes().putAttribute("gtfs_route_short_name", 
-					Normalizer.normalize(routeShortName, Normalizer.Form.NFD).replaceAll("[^\\p{ASCII}]", "")); // replaces non ascii symbols
-		});
-
-		this.convertTrips(activeTrips);
-
-		if(activeTrips.isEmpty()){
-			log.warn("There are no converted trips. You might need to change the date for better results.");
-		}
-		log.info("Conversion successfull");
-	}
-	
-	
-	private void convertStops(){
-		for(Stop stop: feed.stops.values()){
-			TransitStopFacility t = this.ts.getFactory().createTransitStopFacility(Id.create(stop.stop_id, TransitStopFacility.class), transform.transform(new Coord(stop.stop_lon, stop.stop_lat)), false);
-			t.setName(stop.stop_name);
-			ts.addStopFacility(t);
-		}		
-	}
-
-
-	private List<String> getActiveServiceIds(Map<String, Service> services) {
-		List<String> serviceIds = new ArrayList<>();
-		log.info("Used Date for active schedules: " + this.date.toString() + " (weekday: " + date.getDayOfWeek().toString() + "). If you want to choose another date, please specify it, before running the converter");
-		for(Service service: services.values()){
-			if(service.activeOn(date)){
-				serviceIds.add(service.service_id);
-			}
-		}
-		return serviceIds;
-	}
-
-
-	private void convertTrips(List<Trip> trips) {
-		int scheduleDepartures = 0;
-		int frequencyDepartures = 0;
-		for (Trip trip : trips) {
-			if (feed.getFrequencies(trip.trip_id).isEmpty()) {
-				if (feed.getOrderedStopTimesForTrip(trip.trip_id) == null || !feed.getOrderedStopTimesForTrip(trip.trip_id).iterator().hasNext()) {
-					log.error("Found a trip with neither frequency nor ordered stop times. Will not add any Matsim TransitRoute/Departure for that trip. GTFS trip_id=" + trip.trip_id);
-					continue;
-				}
-				StopTime firstStopTime = feed.getOrderedStopTimesForTrip(trip.trip_id).iterator().next();
-				Double departureTime = Time.parseTime(String.valueOf(firstStopTime.departure_time));
-				List<TransitRouteStop> stops = new ArrayList<>();
-				try {
-					for(StopTime stopTime : feed.getInterpolatedStopTimesForTrip(trip.trip_id)) {
-						Id<TransitStopFacility> stopId = Id.create(stopTime.stop_id, TransitStopFacility.class);
-						TransitStopFacility stop = ts.getFacilities().get(stopId);
+
+    private static final Logger log = LogManager.getLogger(GtfsConverter.class);
+
+    private final GTFSFeed feed;
+    private final CoordinateTransformation transform;
+    private final TransitSchedule ts;
+    private final Predicate<Trip> filterTrips;
```

Comment:
I would suggest to rename the predicates. This way one could think that one can set 'if' trips are filtered. I think 'tripFilter', 'stopFilter', would be clearer.

While thinking about it. Maybe something like 'includeTrip' or 'excludeTrip' (don't know what it does) would be even better so that users know in which case they should return true